peachjam/views/generic_views.py

```diff
@@ -20,6 +20,13 @@
 )
 
 
+class BaseQuerysetPrefetchMixin:
+    model = None
+
+    def get_base_queryset(self):
+        return self.model.objects.prefetch_related("author", "nature")
```

Comment:
You could also just set the queryset property on the two views impacted by this; two lines of code instead of a bunch. I also like that the prefetch then is clearly linked to the model

Eg

queryset = GenericDocument.objects.prefetch_related(...)

Up to you

Reply:
I've opted for your suggestion.